cdap-proto/src/main/java/io/cdap/cdap/proto/operation/OperationRunStatus.java

```diff
@@ -48,6 +48,8 @@ public boolean canTransitionTo(OperationRunStatus status) {
       return true;
     }
     switch (this) {
+      case PENDING:
+        return status == STARTING || status == STOPPING;
```

Comment:
can it ever go directly to failed?

Reply:
No the failure message is only send for runtime/runner after STARTING message is processed.